File: bonus_content/tik_tok_effect_for_children/tik_tok_effect_src/Emoji Puzzles 4 Kids/Graph/ScriptTemplate/IfElse/CGIfElse.js
/**
 * @file CGIfElse.js
 * @author
 * @date 2021/8/15
 * @brief CGIfElse.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGIfElse extends BaseNode {
  constructor() {
    super();
  }

  execute() {
    if (true === this.inputs[1]()) {
      if (this.nexts[0]) {
        this.nexts[0]();
      }
    } else {
      if (this.nexts[1]) {
        this.nexts[1]();
      }
    }
  }
}

exports.CGIfElse = CGIfElse;

File: bonus_content/tik_tok_effect_for_children/tik_tok_effect_src/Emoji Puzzles 4 Kids/Graph/ScriptTemplate/IfElse/CGIfElse.lua
local CGIfElse = CGIfElse or {}
CGIfElse.__index = CGIfElse

function CGIfElse.new()
    local self = setmetatable({}, CGIfElse)
    self.inputs = {}
    self.nexts = {}
    return self
end

function CGIfElse:setNext(index, func)
    self.nexts[index] = func
end

function CGIfElse:setInput(index, func)
    self.inputs[index] = func
end

function CGIfElse:execute()
    if self.inputs[1]() then
        if self.nexts[0] then
            self.nexts[0]()
        end
    else
        if self.nexts[1] then
            self.nexts[1]()
        end
    end
end

return CGIfElse
